pandas/io/gbq.py

```diff
@@ -79,23 +87,158 @@ class InvalidColumnOrder(PandasError, IOError):
+    """
+    pass
+
+
+class ResourceInUse(PandasError, ValueError):
+    """
+    Raised when you try to delete a dataset that contains tables or when
+    you try to delete a job that is currently running.
+    """
+    pass
+
+
+class ResourcesExceeded(PandasError, ValueError):
+    """
+    Raised when BigQuery determines that your query uses too many resources.
+    """
+    pass
+
+
+class ResponseTooLarge(PandasError, ValueError):
+    """
+    Raised when query's results are larger than the maximum response size.
+    See `BigQuery quota policy <https://cloud.google.com/bigquery/quota-policy#queries>`__
+
+    BigQuery states that if you want to ensure that large results can return, you can set the
+    allowLargeResults property to true and specify a destination table. This feature is not
+    currently supported in the pandas gbq module.
+    See `Issue #10474 <https://github.com/pydata/pandas/issues/10474>`__
+    """
+    pass
+
+
+class StreamingInsertError(PandasError, ValueError):
+    """
+    Raised when BigQuery reports a streaming insert error.
+    For more information see `Streaming Data Into BigQuery <https://cloud.google.com/bigquery/streaming-data-into-bigquery>`__
+    """
+
+
+class TableAlreadyExists(PandasError, ValueError):
+    """
+    Raised when the create table method finds the
+    desired table already exists
+    """
+    pass
+
+
+class TableCreationError(PandasError, ValueError):
+    """
+    Raised when the create table method fails
+    """
+    pass
+
+
+class TermsOfServiceNotAccepted(PandasError, ValueError):
+    """
+    Raised when the terms of service were not accepted.
+    """
+    pass
+
+class UnknownGBQException(PandasError, ValueError):
+    """
+    Raised when an unrecognized Google API Error occurs.
+    """
+    pass
+
+
 class GbqConnector(object):
 
     def __init__(self, project_id, reauth=False):
+        self.test_google_api_imports()
+        self.project_id = project_id
+        self.reauth = reauth
+        self.credentials = self.get_credentials()
+        self.service = self.get_service(self.credentials)
 
-        self.project_id     = project_id
-        self.reauth         = reauth
-        self.credentials    = self.get_credentials()
-        self.service        = self.get_service(self.credentials)
-
-    def get_credentials(self):
+    def test_google_api_imports(self):
         try:
+            import httplib2
+            from apiclient.discovery import build
+            from apiclient.errors import HttpError
+            from oauth2client.client import AccessTokenRefreshError
             from oauth2client.client import OAuth2WebServerFlow
             from oauth2client.file import Storage
             from oauth2client.tools import run_flow, argparser
-
         except ImportError:
-            raise ImportError('Could not import Google API Client.')
+            raise ImportError('Missing required module')
```

Comment:
do ``except ImportError as e:`` ,then raise something like ``ImportError("......gbq: {0}".format(str(e)))`` (e.g. you need to re-raise, but with a more informative message